savant/converter/yolo.py

```diff
@@ -34,6 +39,50 @@ def __init__(
         self.class_ids = class_ids
         super().__init__()
 
+    def _decode_single_head(
+        self, output: np.ndarray, num_classes: Optional[int]
+    ) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """
+        Returns (bboxes_xywh, confidences, class_ids) from a single output tensor.
+        Handles both [N, 4+C]/[N, 5+C] and transposed [(4+C),N]/[(5+C),N].
+        """
+        # Squeeze batch dim if present
+        if output.ndim == 3 and output.shape[0] == 1:
+            output = output[0]
+
+        # Try to detect transposed layout using num_classes if provided
+        def _is_transposed(arr: np.ndarray) -> bool:
+            if num_classes is None:
+                # Heuristic: if first dim is "plausible" channel count (84/85/etc.)
+                return arr.shape[0] in (84, 85, 92, 93)  # typical for COCO and similar
```

Comment:
I believe that we cannot use such heuristics. If there is no way to automatically determine the YOLO type, let's introduce the type as a mandatory argument for the converter.